cmd/jaeger/internal/integration/badger_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/integration/storagecleaner"
+	"github.com/jaegertracing/jaeger/plugin/storage/integration"
+)
+
+type BadgerStorageIntegration struct {
+	E2EStorageIntegration
+	logger *zap.Logger
+}
+
+func (s *BadgerStorageIntegration) initialize(t *testing.T) {
+	s.e2eInitialize(t)
+
+	s.CleanUp = s.cleanUp
+	s.logger = zap.NewNop()
+}
+
+func (s *BadgerStorageIntegration) cleanUp(t *testing.T) {
+	addr := fmt.Sprintf("http://0.0.0.0:%s%s", storagecleaner.Port, storagecleaner.URL)
+	r, err := http.NewRequest(http.MethodPost, Addr, nil)
+	require.NoError(t, err)
+
+	client := &http.Client{}
+
+	resp, err := client.Do(r)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	require.Equal(t, http.StatusOK, resp.StatusCode)
+}
+
+func TestBadgerStorage(t *testing.T) {
+	integration.SkipUnlessEnv(t, "badger")
+
+	s := &BadgerStorageIntegration{
+		E2EStorageIntegration: E2EStorageIntegration{
+			ConfigFile: createBadgerCleanerConfig(t),
+			StorageIntegration: integration.StorageIntegration{
+				SkipBinaryAttrs: true,
+				SkipArchiveTest: true,
+
+				// TODO: remove this once badger supports returning spanKind from GetOperations
+				// Cf https://github.com/jaegertracing/jaeger/issues/1922
+				GetOperationsMissingSpanKind: true,
+			},
+		},
+	}
+	s.initialize(t)
+	t.Cleanup(func() {
+		s.e2eCleanUp(t)
+	})
+	s.RunAll(t)
+}
+
+func createBadgerCleanerConfig(t *testing.T) string {
```

Comment:
suggest moving to integration_test.go as `createStorageCleanerConfig`, since we may call it from ES tests as well.

In fact, I don't think we should be calling this function explicitly from Badger test at all, just pass the badger config as property to `E2EStorageIntegration` and let it create a temp file with extension (regardless of whether the specific storage test needs it or not). Less code / initialization this way.